Content.Shared/Chameleon/Components/SharedChameleonComponent.cs

```diff
@@ -8,46 +8,52 @@ namespace Content.Shared.Chameleon.Components;
 /// It also turns the entity invisible
 /// </summary>
 [RegisterComponent, NetworkedComponent]
+[Access(typeof(SharedChameleonSystem))]
 public sealed class SharedChameleonComponent : Component
 {
     /// <summary>
-    ///     Whether or not the entity previously had an interaction outline prior to cloaking.
+    /// Whether or not the entity previously had an interaction outline prior to cloaking.
     /// </summary>
     [DataField("hadOutline")]
     public bool HadOutline;
 
     /// <summary>
-    /// Current level of stealth based on movement in the <exception cref="Chameleon SYstem"></exception>
-    /// -1f and 1f are the current maximums for now.
+    /// Last set level of visibility. Ranges from 1 (fully visible) and -1 (fully hidden). To get the actual current
+    /// visibility, use <see cref="SharedChameleonSystem.Getvisibility(EntityUid, SharedChameleonComponent?)"/>
     /// </summary>
-    [ViewVariables]
-    [DataField("stealthLevel")]
-    public float StealthLevel;
+    [DataField("lastVisibility")]
+    [Access(typeof(SharedChameleonSystem),  Other = AccessPermissions.None)]
```

Comment:
The `Other = AccessPermissions.None` is to ensure that other systems go through `SharedChameleonSystem.GetVisibility()`, so that the elapsed time is properly included.